### PR TITLE
Experimental/flag options

### PR DESCRIPTION
**Pull request by ens-bwalts**

### CODE REVIEW

File: scripts/beekeeper.pl [hunk 1]
```diff
@@ -422,13 +422,29 @@ sub generate_worker_cmd {
...
+            }
+        }
+    }
+
+    # nosqlvc is a special case for compatibility with modules where it is a propigated option
+    if ((defined($self->{nosqlvc})) &&
+        ($self->{nosqlvc} == 1)) {
+        $worker_cmd .= " -nosqlvc";
+    }
+
```
Comment (muffato) [inline scripts/beekeeper.pl]: I think `nosqlvc` could be covered by the _foreach_ loop just above as it is either undefined or set to 1

File: scripts/runWorker.pl [hunk 2]
```diff
@@ -67,8 +67,8 @@ sub main {
                'worker_cur_dir|cwd=s'       => \$worker_cur_dir,
                'hive_log_dir|hive_output_dir=s'         => \$hive_log_dir,       # keep compatibility with the old name
                'worker_log_dir|worker_output_dir=s'     => \$worker_log_dir,     # will take precedence over hive_log_dir if set
-               'retry_throwing_jobs=i'      => \$retry_throwing_jobs,
-               'can_respecialize=i'         => \$can_respecialize,
+               'retry_throwing_jobs!'       => \$retry_throwing_jobs,
```
Comment (muffato) [inline scripts/runWorker.pl]: You'll have to change `-retry_throwing_jobs 1` in docs/running_pipelines/running.rst

File: scripts/runWorker.pl [hunk 3]
```diff
@@ -56,7 +56,7 @@ sub main {
                'analysis_id=i'              => \$analysis_id,
                'logic_name=s'               => \$logic_name,
                'job_id=i'                   => \$job_id,
-               'force=i'                    => \$force,
+               'force'                      => \$force,
```
Comment (muffato) [inline scripts/runWorker.pl]: You'll have to change `-force 1` in docs/running_pipelines/running.rst

Comment (muffato) [reply scripts/runWorker.pl]: There are also several `-force 1` in docs/running_pipelines/troubleshooting.rst
BTW [troubleshooting.rst](http://ensembl-hive.readthedocs.io/en/master/running_pipelines/troubleshooting.html#the-runworker-pl-script) overlaps [running.rst](http://ensembl-hive.readthedocs.io/en/master/running_pipelines/running.html#running-a-single-job-using-runworker-pl) quite a lot

File: docs/running_pipelines/management.rst [hunk 4]
```diff
@@ -272,18 +272,6 @@ for new jobs that it can run, and will claim and run them automatically
...
-can claim and work on jobs of a different analysis when needed. If you
-run with ``-can_respecialize 1``, you may notice a significant number of
-workers used throughout the pipeline as long as different analyses share
-the same resource classes (i.e. the resource classes are not too specific).
-For instance, if all the analyses are linked to the same resource-class, a
-single worker in ``-can_respecialize 1`` mode would be able to run the
-whole pipeline !
-Since fewer workers are submitted, the general responsiveness of the
-compute cluster and the pipeline are also increased.
-
```
Comment (muffato) [inline docs/running_pipelines/management.rst]: Do we explain somewhere the life cycle of a worker ? That would need to mention that by default, _beekeeper_ submits workers that can respecialize (and what respecialization means)

File: scripts/beekeeper.pl [hunk 5]
```diff
@@ -439,8 +455,14 @@ sub generate_worker_cmd {
...
 
+    # force is given as an argument to the sub, rather than being taken from the options hash
     if (defined($force)) {
-        $worker_cmd .= " -force $force";
+        # ensure a -noforce is passed through in the unlikely case where it is set
+        if ($force == 0) {
+            $worker_cmd .= "-noforce";
+        } else {
+            $worker_cmd .= " -force";
+        }
```
[the remark this answers is not in the document]
Comment (ens-bwalts) [reply scripts/beekeeper.pl]: I don't see any reason why it couldn't be handled as a script-wide parameter. Do you happen to know why it gets passed as an argument to run_autonomously() and generate_worker_command() - unlike most of the other options? I can't see a reason for handling it differently in the code as it stands now.

[the remark this answers is not in the document]
Comment (muffato) [reply scripts/beekeeper.pl]: No I don't know, and I don't see any reasons either

**Comment by muffato** [comment]
Following up our discussion: add `can_respecialise` as an alias to `can_respecialize`

**Pull request merged**